fix: update readme (#76)
terraform-ibm-modules/terraform-ibm-hpcs: README.md

For more information, see [components and concepts](https://cloud.ibm.com/docs/hs-crypto?topic=hs-crypto-understand-concepts) of HPCS and [about service instance initialization](https://cloud.ibm.com/docs/hs-crypto?topic=hs-crypto-introduce-service) in the Cloud Docs.

> If you provision HPCS instance with `private-only` endpoint then you receive the `public` and `private` endpoints URL in the output. Only the private endpoint works. The public endpoint is just for reference and only valid if public-and-private endpoint is setup.

Reviewer: I think this should go below in the readme as opposed to being one of the first thing a new consumer to the module sees. Could we collocate this with the private variable doc?

> If you provision HPCS instance with `private-only` endpoint then you receive the `public` and `private` endpoints URL in the output. Only the private endpoint works. The public endpoint is just for reference and only valid if public-and-private endpoint is setup.

Reviewer 2: ```suggestion
If you provision an HPCS instance with a `private-only` endpoint, both `public` and `private` endpoints URLs are included in the output. You can disregard the public endpoint.
```

I'm assuming there's no real reason for the public endpoint in this situation. If there is, can we be specific about why it's included? What does the public endpoint refer to if it's "just for reference"?

Author: Added some more details. Please review if that helps.

For more information, see [components and concepts](https://cloud.ibm.com/docs/hs-crypto?topic=hs-crypto-understand-concepts) of HPCS and [about service instance initialization](https://cloud.ibm.com/docs/hs-crypto?topic=hs-crypto-introduce-service) in the Cloud Docs.

> If you provision an HPCS instance with a `private-only` endpoint, both `public` and `private` endpoints URLs are included in the output. The public endpoint can be referenced only if there is a need for a temporary switch to enable public endpoint otherwise you can disregard the public endpoint.

Reviewer 2: ```suggestion
If you provision an HPCS instance with a `private-only` endpoint, both `public` and `private` endpoints URLs are included in the output. You can switch to the public endpoint if needed, but generally you can ignore it.
```

See if you think this is clearer.

Author: Hmm, I think if we say directly that `you can switch...` then it generates some more questions related to security.  

Following are the exact words from HPCS team if that helps us to reframe it better.
```
we should document the public endpoint references are there for convenience reason and only valid 
if public-and-private is setup. There might be usecases where it was setup as private-only, and for 
a temporary switch to enable public, the endpoint info is already there for applications to be prepared
 start using it, even before the config change....
```

Author: Can we say this - 
```
If you provision an HPCS instance with a `private-only` endpoint, both `public` and `private` endpoints
URLs are included in the output. The public endpoint can be used if there is a need for a temporary 
switch to enable public endpoint without any configuration change otherwise you can ignore it.
```

Reviewer 2: Ah, I get it. We want to try to use active voice as much as possible (for example, not, `can be used`, but `you can use`). Here's a take with a cautionary note. But go with what you think works best. 

> If you provision an HPCS instance with a `private-only` endpoint, both `public` and `private` endpoints URLs are included in the output. The public endpoint is included for convenience in case you need to switch to it temporarily. However, make sure that you switch back to the private endpoint as soon as possible.

Author: Thanks a lot Allen. Made changes like this:
```
If you provision an HPCS instance with a private-only endpoint, both public and private endpoints 
URLs are included in the output. You can ignore the public endpoint. It is included for convenience
 in case you need to switch to it temporarily. However, make sure that you switch back to the 
private endpoint as soon as possible.
```
